upgrade-2.x.sh

```diff
@@ -352,26 +363,44 @@ function upgrade_supervisor() {
         fi
     fi
 
-    if CURRENT_SUPERVISOR_VERSION=$(_fetch_supervisor_version); then
-        if version_gt "$target_supervisor_version" "$CURRENT_SUPERVISOR_VERSION" ; then
-            log "Supervisor update: will be upgrading from v${CURRENT_SUPERVISOR_VERSION} to v${target_supervisor_version}"
-            progress 90 "Running supervisor update"
-            if _patch_supervisor_version "$target_supervisor_version"; then
-                # No need to stop services as supervisor update will do that for us
-                if _run_supervisor_update; then
-                    if version_gt "6.5.9" "${target_supervisor_version}" ; then
-                        remove_containers
-                        log "Removing supervisor database for migration"
-                        rm /resin-data/resin-supervisor/database.sqlite || true
-                    fi
-                else
-                    log WARN "Failed to update supervisor to target state version - leave to first boot."
+    if supervisor_target_state_versions=$(_fetch_supervisor_version); then
```

Comment:
is this really a valid if condition in bash? It's not actually checking that they are equal, just that the subshell did not fail, correct?

Reply:
That's right @klutchell it's checking the return code of the _fetch_supervisor_version.